src/commands/api/deployWorkspaceProjectApi.ts

```diff
@@ -0,0 +1,66 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { type ResourceGroup } from "@azure/arm-resources";
+import { ResourceGroupListStep, parseAzureResourceGroupId } from "@microsoft/vscode-azext-azureutils";
+import { createSubscriptionContext, subscriptionExperience, type IActionContext, type ISubscriptionActionContext, type ISubscriptionContext } from "@microsoft/vscode-azext-utils";
+import { type AzureSubscription } from "@microsoft/vscode-azureresources-api";
+import { Uri, type WorkspaceFolder } from "vscode";
+import { ext } from "../../extensionVariables";
+import { getWorkspaceFolderFromPath } from "../../utils/workspaceUtils";
+import { deployWorkspaceProjectInternal, type DeployWorkspaceProjectInternalContext } from "../deployWorkspaceProject/deployWorkspaceProjectInternal";
+import { getDeployWorkspaceProjectResults, type DeployWorkspaceProjectResults } from "../deployWorkspaceProject/getDeployWorkspaceProjectResults";
+import type * as api from "./vscode-azurecontainerapps.api";
+
+export async function deployWorkspaceProjectApi(context: IActionContext, deployWorkspaceProjectOptions: api.DeployWorkspaceProjectOptionsContract): Promise<DeployWorkspaceProjectResults> {
+    const { resourceGroupId, rootPath, dockerfilePath, srcPath, suppressConfirmation, suppressContainerAppCreation, ignoreExistingDeploySettings, shouldSaveDeploySettings } = deployWorkspaceProjectOptions;
+
+    const subscription: AzureSubscription = await subscriptionExperience(context, ext.rgApiV2.resources.azureResourceTreeDataProvider, {
+        selectBySubscriptionId: getSubscriptionIdFromOptions(deployWorkspaceProjectOptions),
+        showLoadingPrompt: false
+    });
+    const subscriptionContext: ISubscriptionContext = createSubscriptionContext(subscription);
+
+    const rootFolder: WorkspaceFolder | undefined = rootPath ? getWorkspaceFolderFromPath(rootPath) : undefined;
+    const resourceGroup: ResourceGroup | undefined = resourceGroupId ? await getResourceGroupFromId({ ...context, ...subscriptionContext }, resourceGroupId) : undefined;
+
+    const deployWorkspaceProjectInternalContext: DeployWorkspaceProjectInternalContext = Object.assign(context, {
+        ...subscriptionContext,
+        subscription,
+        resourceGroup,
+        rootFolder,
+        srcPath: srcPath ? Uri.file(srcPath).fsPath : undefined,
```

Comment:
> Isn't srcPath already the fsPath of the Uri?

It's supposed to be, but I don't think there's anything guaranteeing that the person passing the option gives the correct path option.  I figured that if someone accidentally passed `.path` as an argument through the api option, this would normalize it back to the correct `.fsPath` for the machine.  Let me know if that makes sense.

Reply:
> Side tangent: Is it an URI or a URI?

That makes:

`a`: 2
`an`: 0

Reply:
Yeah, it makes sense. I doubt it's very expensive to create a `Uri` object, but it just seemed needless. But yeah, I can understand wanting to ensure that we are standardizing the fsPath